### PR TITLE
fix(ui): context initials skip generated-id segments and use up to three letters

### PR DESCRIPTION
**Pull request by deveshk0**
Relates to #209 — the abbreviation-quality half (the hotbar overflow half landed in #211). Left to the issue reporter/maintainer to close after verifying against their real context set.

## What

`avatarInitials` took the first letter of the first two segments, which collapsed the issue's real-world contexts into identical labels:

| context | before | after |
|---|---|---|
| `dev-lon-nrtc-6bcb8b63` | DL | **DLN** |
| `dev-lon-nrtc-6bcb8b63-admin` | DL | **DLN** |
| `dev-lon-workload-15d9c530` | DL | **DLW** |
| `mbr-lon-nrtc-c678b415` | ML | **MLN** |
| `k3d-dev-ai` | KD | **KDA** |
| `k3d-nats` | KN | KN |

Initials now come from up to **three** segments, **skipping segments that carry no identity**: hex ids (6+ chars, required to contain a digit so real hex-alphabet words like `decade` still count) and bare numbers. A name that is entirely generated ids falls back to the raw segments so it always gets a label.

The `-admin` twins still share a label by construction — they differ only by their id — but they keep distinct avatar colours (hashed from the full name), the full name in the tooltip, and the per-context short-name override in Preferences remains the explicit escape hatch.

## Testing

Unit tests over the issue's real context set, the hex-word/mixed-segment edge cases, and the all-generated fallback. Full frontend suite green (991).

### CODE REVIEW

File: apps/desktop/src/ui/avatar.ts [hunk 1]
```diff
@@ -29,10 +29,30 @@ export function avatarColor(name: string): string {
...
 }
 
-/** Up-to-2-char initials from a cluster name (splits on - _ space /). */
+/** A segment that carries no identity: a generated id (hex with at least one
+ * digit, 6+ chars — `6bcb8b63`) or a bare number. Cluster names commonly end
+ * in these, and their initial tells the user nothing. The digit requirement
+ * keeps real words that happen to be hex-alphabet (`decade`, `facade`) out
+ * of the noise class. */
+function isNoiseSegment(segment: string): boolean {
+  return /^(?=.*\d)[0-9a-f]{6,}$/i.test(segment) || /^\d+$/.test(segment);
```
[the remark this answers is not in the document]
Comment (deveshk0) [reply apps/desktop/src/ui/avatar.ts]: Fixed in `3159110`: the id heuristic now requires 8+ hex chars (plus the existing at-least-one-digit rule), so 6-char leetspeak-style names (`prod-c0ffee` → PC, `prod-babe42` → PB — both test cases now) keep their identity segment while the real generated ids in the #209 set (all 8 hex) are still skipped. Position-based constraints were considered and rejected: the `...-6bcb8b63-admin` twins carry the id mid-name, so a trailing-only rule would miss them.

File: apps/desktop/src/ui/avatar.ts [hunk 2]
```diff
@@ -29,10 +29,32 @@ export function avatarColor(name: string): string {
...
-/** Up-to-2-char initials from a cluster name (splits on - _ space /). */
+/** A segment that carries no identity: a generated id (hex with at least one
+ * digit, 8+ chars — `6bcb8b63`) or a bare number. Cluster names commonly end
+ * in these, and their initial tells the user nothing. Deliberately narrow,
+ * since segments are user-defined: the digit requirement keeps hex-alphabet
+ * words (`decade`, `cafebabe`) as identity, and the 8-char floor keeps
+ * short leetspeak-style names (`c0ffee`, `babe42`) as identity too — real
+ * generated ids (k3d, EKS, the #209 set) are 8 hex chars or more. */
+function isNoiseSegment(segment: string): boolean {
+  return /^(?=.*\d)[0-9a-f]{8,}$/i.test(segment) || /^\d+$/.test(segment);
```
[the remark this answers is not in the document]
Comment (deveshk0) [reply apps/desktop/src/ui/avatar.ts]: Fixed in `8ea9e63`: bare numbers count as noise only at 6+ digits (timestamp/serial shapes), so `cluster-1`/`cluster-2` keep their distinguishing digit (C1/C2 — both test cases now) while `prod-eu-20260813` still skips the date. Short numbers are usually the whole identity of a numbered cluster; the heuristic now only ever drops digit runs no one would type as a name.
